tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py

```diff
@@ -166,21 +171,22 @@ def _process_ready_predictions(batch_raw_predictions, batch_identifiers, batch_m
                 time.sleep(wait_time)
                 wait_time = max(wait_time * 2, .16)
 
-        if dump_prediction_to_annotation and self._dumped_annotations:
-            self.register_annotations(self._dumped_annotations)
+        if dump_prediction_to_annotation:
+            self.register_dumped_annotations()
 
         if progress_reporter:
             progress_reporter.finish()
 
-    def register_annotations(self, annotations):
-        annotation_reader = Dataset(self.dataset.dataset_config, True)
-        annotation_reader.set_annotation(annotations)
-        self.dataset.annotation_reader = annotation_reader
+    def register_dumped_annotations(self):
+        if not self._dumped_annotations:
+            return
+        if self.dataset.annotation_reader is None:
+            self.dataset.annotation_reader = Dataset(self.dataset.dataset_config, True)
+        self.dataset.annotation_reader.set_annotation(self._dumped_annotations)
 
     def select_dataset(self, dataset_tag):
         if self.dataset is not None and isinstance(self.dataset_config, list):
-            if self.dataset.annotation_reader is None and self._dumped_annotations:
-                self.register_annotations(self._dumped_annotations)
+            self.register_dumped_annotations()
```

Comment:
you move dumping to the process, so I not sure that you need it here, you just convert it twice.

Reply:
Will test it now